tls/s2n_config.c

```diff
@@ -846,11 +846,12 @@ int s2n_config_enable_cert_req_dss_legacy_compat(struct s2n_config *config)
     return S2N_SUCCESS;
 }
 
-int s2n_config_set_psk_selection_callback(struct s2n_config *config, s2n_psk_selection_callback cb)
+int s2n_config_set_psk_selection_callback(struct s2n_config *config, s2n_psk_selection_callback cb, void *context)
 {
     POSIX_ENSURE_REF(config);
     POSIX_ENSURE_REF(cb);
```

Comment:
I forgot to mention this on the other pr: we need to be able to unset callbacks so we can't ensure ref this

Reply:
Fair. Want to add that to the guidelines somehow? I know we've commented that a few APIs now

Reply:
It is a little odd to use the `set_psk_selection_callback` to unset the `cb` as well. Do we want to document it or change the name?